opentelemetry-sdk/src/metrics/data/mod.rs

```diff
@@ -97,7 +97,7 @@ impl<T: fmt::Debug + Send + Sync + 'static> Aggregation for Sum<T> {
 pub struct DataPoint<T> {
     /// Attributes is the set of key value pairs that uniquely identify the
     /// time series.
-    pub attributes: AttributeSet,
+    pub attributes: Vec<KeyValue>,
```

Comment:
Could we use a boxed slice instead of a Vec? It would save 8 bytes (`usize` length) per instance since unlike we wouldn't need an additional pointer to track the capacity. That could save a considerable amount of memory if we are going to have a lot of these `DataPoints`.

```suggestion
    pub attributes: Box<[KeyValue]>,
```

Reply:
As mentioned in the PR desc, will revisit the public APIs. This PR is to unblock the unwanted exposure of `AttributeSet` outside of core sdk.